src/main/java/ama/post/PostController.java

```diff
@@ -41,11 +40,17 @@ public String createAMASubmission(Model model) {
     @PostMapping("/create_submission")
     public String postAMASubmission(@ModelAttribute("submissionPost") SubmissionPost post, BindingResult result){
         validator.validate(post, result);
-        if(result.hasErrors()) {
+
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        String username = auth.getName();
+        User user = userRepository.findByUsername(username);
+        if(result.hasErrors()||user == null) {
```

Comment:
I'm guessing this takes care of the case if a non-registered user tried to make an AMA?